tests/integration_tests/test_cli.py

```diff
@@ -300,3 +301,86 @@ def test_preprocess_cli(csv_filename):
         config_filename = os.path.join(tmpdir, "config.yaml")
         dataset_filename = _prepare_data(csv_filename, config_filename)
         _run_ludwig("preprocess", dataset=dataset_filename, preprocessing_config=config_filename)
+
+
+@pytest.mark.distributed
+@pytest.mark.parametrize("second_seed_offset", [0, 1])
+@pytest.mark.parametrize("random_seed", [1919, 31])
+@pytest.mark.parametrize("type_of_run", ["train", "experiment"])
+@pytest.mark.parametrize("backend", ["local", "horovod"])
+def test_reproducible_cli_runs(
+    backend: str, type_of_run: str, csv_filename: str, random_seed: int, second_seed_offset: int
+) -> None:
+    """
+    Test for reproducible training using `ludwig experiment|train --dataset`.
+    Args:
+        backend (str): backend to use
+        type_of_run(str): type of run, either train or experiment
+        csv_filename(str): file path of dataset to use
+        random_seed(int): random seed integer to use for test
+        second_seed_offset(int): zero to use same random seed for second test, non-zero to use a different
+            seed for the second run.
+
+    Returns: None
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Instead of setting up a `tmpdir` manually, tmpdir is a built-in fixture of pytest:

https://github.com/ludwig-ai/ludwig/blob/master/tests/integration_tests/test_api.py#L188

Reply:
converted to use pytest fixture `tmpdir`.